CometServer/Services/Operations/SideEffects/Implementation/ParameterOverrideValueSetSideEffect.cs

```diff
@@ -72,5 +84,53 @@ public override void BeforeDelete(ParameterOverrideValueSet thing, Thing contain
         {
             throw new InvalidOperationException("ParameterOverrideValueSet Cannot be deleted");
         }
+
+        /// <summary>
+        /// Allows derived classes to override and execute additional logic before an update operation.
+        /// </summary>
+        /// <param name="thing">
+        /// The <see cref="Thing"/> instance that will be inspected.
+        /// </param>
+        /// <param name="container">
+        /// The container instance of the <see cref="Thing"/> that is inspected.
+        /// </param>
+        /// <param name="transaction">
+        /// The current transaction to the database.
+        /// </param>
+        /// <param name="partition">
+        /// The database partition (schema) where the requested resource will be stored.
+        /// </param>
+        /// <param name="securityContext">
+        /// The security Context used for permission checking.
+        /// </param>
+        /// <param name="rawUpdateInfo">
+        /// The raw update info that was serialized from the user posted request.
+        /// The <see cref="ClasslessDTO"/> instance only contains values for properties that are to be updated.
+        /// It is important to note that this variable is not to be changed likely as it can/will change the operation processor outcome.
+        /// </param>
+        public override void BeforeUpdate(ParameterOverrideValueSet thing, Thing container, NpgsqlTransaction transaction, string partition, ISecurityContext securityContext, ClasslessDTO rawUpdateInfo)
+        {
+            base.BeforeUpdate(thing, container, transaction, partition, securityContext, rawUpdateInfo);
+
+            if (container is not ParameterOverride parameterOverride)
+            {
+                throw new InvalidOperationException("The container of the ParameterOverrideValueSet is not a ParameterOverride");
+            }
+
+            var parameter = this.ParameterService.Get(transaction, partition, new List<Guid> { parameterOverride.Parameter }, securityContext)
```

Comment:
i would like that this becomes a bit smarter and we only do this when at least one of the following is true: The rawupdateinfo contains and `reference`, `manual`, `computed` update.

only then do we need to validate and hence collect all the required data to perform validation. In case the user only updates the switch or actualstate or actualoption we would not need to do all this work